src/schemes/fluid/viscosity.jl

```diff
@@ -197,36 +224,232 @@ end
     v_b = viscous_velocity(v_neighbor_system, neighbor_system, neighbor)
     v_diff = v_a - v_b
 
-    eta_a = nu_a * rho_a
-    eta_b = nu_b * rho_b
+    return adami_viscosity_force(smoothing_length_average, pos_diff, distance, grad_kernel,
+                                 m_a, m_b, rho_a, rho_b, v_diff, nu_a, nu_b, epsilon)
+end
 
-    eta_tilde = 2 * (eta_a * eta_b) / (eta_a + eta_b)
+function kinematic_viscosity(system, viscosity::ViscosityAdami, smoothing_length,
+                             sound_speed)
+    return viscosity.nu
+end
+
+@propagate_inbounds function viscous_velocity(v, system, particle)
+    return current_velocity(v, system, particle)
+end
+
+@doc raw"""
+    ViscosityAdamiSGS(; nu, C_S=0.1, epsilon=0.01)
+
+Viscosity model that extends the standard [Adami formulation](@ref ViscosityAdami)
+by incorporating a subgrid-scale (SGS) eddy viscosity via a Smagorinsky-type [Smagorinsky (1963)](@cite Smagorinsky1963) closure.
+The effective kinematic viscosity is defined as
+
+```math
+\nu_{\mathrm{eff}} = \nu_{\\mathrm{std}} + \nu_{\\mathrm{SGS}},
+```
+
+with
+
+```math
+\nu_{\mathrm{SGS}} = (C_S * h)^2 * |S|,
+```
+
+and an approximation for the strain rate magnitude given by
+
+```math
+|S| \approx \frac{\|v_a - v_b\|}{\|r_a - r_b\| + \epsilon},
```

Comment:
```suggestion
|S| \approx \frac{\|v_ab\|}{\|r_ab\| + \epsilon},
```
I think we used this everywhere else.